### PR TITLE
EVG-8064 add flag to check if user has used spruce before

### PR DESCRIPTION
**Pull request by khelif96**
Didnt add a test for this flag since the functionality is covered under the user-settings tests.

### CODE REVIEW

File: graphql/schema.graphql [hunk 1]
```diff
@@ -119,6 +119,7 @@ input UserSettingsInput {
   githubUser: GithubUserInput
   slackUsername: String
   notifications: NotificationsInput
+  useSpruceOptions: UseSpruceOptionsInput
```
Comment (john-m-liu) [inline graphql/schema.graphql:122]: What are you planning to use this parameter for? I would think that it's just something that gets returned but I'm probably missing something

File: graphql/schema.graphql [hunk 2]
```diff
@@ -431,6 +436,11 @@ type UserSettings {
   githubUser: GithubUser
   slackUsername: String
   notifications: Notifications
+  useSpruceOptions: UseSpruceOptions
+}
+
+type UseSpruceOptions {
+  hasUsedSpruceBefore: Boolean
```
Comment (john-m-liu) [inline graphql/schema.graphql:443]: Boolean! maybe?

File: graphql/schema.graphql (hunk 1 above)
Comment (khelif96) [reply graphql/schema.graphql:122]: When the user views the welcome modal we want to toggle this field so we dont show them the modal more than once.  I will also be using this field for the opt in/out ability of spruce.

Comment (john-m-liu) [reply graphql/schema.graphql:122]: Doh right, wasn't thinking there

**Pull request merged**